Sources/SwiftDocCUtilities/Action/Actions/Convert/ConvertAction.swift

```diff
@@ -462,14 +462,18 @@ public struct ConvertAction: Action, RecreatingContext {
         }
 
         var didEncounterError = analysisProblems.containsErrors || conversionProblems.containsErrors
-        if try context.renderRootModules.isEmpty {
+        if (try context.renderRootModules + context.rootTechnologies).isEmpty {
             postConversionProblems.append(
                 Problem(
                     diagnostic: Diagnostic(
                         severity: .warning,
                         identifier: "org.swift.docc.MissingTechnologyRoot",
-                         summary: "No TechnologyRoot to organize article-only documentation.",
-                         explanation: "Article-only documentation needs a TechnologyRoot page (indicated by a `TechnologyRoot` directive within a `Metadata` directive) to define the root of the documentation hierarchy."
+                         summary: "No TechnologyRoot to organize the documentation.",
+                         explanation: """
+                        There was no root found for this documentation catalog.
+                        - For Article-only documentation please add a TechnologyRoot page (indicated by a `TechnologyRoot` directive within a `Metadata` directive) to define the root of the documentation hierarchy.
+                        - For Tutorials please add a table of contents page (indicated by a `Tutorials` directive with the corresponding `Intro` and `Chapters` directive) to define the root of the documentation hierarchy.\n
```

Comment:
Nice adding the description for tutorial-only documentation here. 

I think we could make the tutorial-only experience even nicer if we checked if the context had any tutorials and used different diagnostics for article-only and tutorial-only content.  

What do you think?

Reply:
Actually, now that I think about this some more... since TechnologyRoot is optional now; this error message is only applicable for tutorial-only documentation, right?

Reply:
> Actually, now that I think about this a bit more... since TechnologyRoot is optional now; this error message is only applicable for tutorial-only documentation, right? So I think that this warning only needs to cover tutorial-only documentation.

I considered this, but since the synthesised technology root behaviour is not documented yet (I think) I was not sure if I wanted to remove the warning.

Reply:
What about this? 
```
 There was no root found for this documentation catalog.

For Tutorials please add a table of contents page (indicated by a `Tutorials` directive with the corresponding `Intro` and `Chapters` directive) to define the root of the documentation hierarchy.
```

Emitting the warning when there's no root could be beneficial to give to the user an idea of the problem independently of the type of catalog they're trying to build.

Reply:
We could possibly do something like 

```swift
let hasTutorial = context.knownPages.contains(where: {
    guard let kind = try? context.entity(with: $0).kind else {
        return false
    }
    return kind == .tutorial || kind == .tutorialArticle
})
```

and customize the error message depending on if we know that the catalog has a tutorial or not.

Reply:
Implemented here https://github.com/apple/swift-docc/pull/812/commits/71cd2a6ac63320a99807f74770aa7fbed7402cbb